q2cli/builtin/dev.py

```diff
@@ -141,57 +143,93 @@ def reset_theme():
         click.echo('Theme was already default.')
 
 
-@dev.command(name='assert-output-type',
-             short_help='lorem ipsum',
-             help='lorem ipsum',
+@dev.command(name='assert-result-type',
+             short_help='Assert Result is a specific type.',
+             help='Checks that the type of a Result matches an '
+                  'expected type. Intended for developer testing.',
              cls=ToolCommand)
-@click.option('--input-path', required=True, metavar='ARTIFACT/VISUALIZATION',
-              type=click.Path(exists=True, file_okay=True,
-                              dir_okay=False, readable=True),
-              help='Lorem Ipsum')
+@click.argument('input-path', type=click.Path(exists=True, file_okay=True,
+                dir_okay=False, readable=True),
+                metavar=_COMBO_METAVAR)
 @click.option('--qiime-type', required=True,
-              help='Lorem Ipsum')
-def assert_output_type(input_path, qiime_type):
+              help='QIIME 2 data type.')
+def assert_result_type(input_path, qiime_type):
     import q2cli.util
     import qiime2.sdk
+    from q2cli.core.config import CONFIG
 
     q2cli.util.get_plugin_manager()
-    result = qiime2.sdk.Result.load(input_path)
-
-    if str(result.type) == qiime_type:
-        click.echo("Good to go")
+    try:
+        result = qiime2.sdk.Result.load(input_path)
+    except Exception as e:
+        header = 'There was a problem loading %s as a QIIME 2 Result:' % \
+            input_path
+        q2cli.util.exit_with_error(e, header=header)
+
+    if str(result.type) != qiime_type:
+        msg = 'Expected %s, observed %s' % (qiime_type, result.type)
+        click.echo(CONFIG.cfg_style('error', msg), err=True)
+        click.get_current_context().exit(1)
     else:
-        raise Exception("Expected %s, observed %s" % (qiime_type,
-                                                      result.type))
+        msg = 'The type of the input file: %s and the expected type: %s' \
+              ' match' % (input_path, qiime_type)
+        click.echo(CONFIG.cfg_style('success', msg))
 
 
-@dev.command(name='assert-has-line',
-             short_help='Checks that provided line is present in input file',
-             help='Lorem Ipsum',
+@dev.command(name='assert-result-data',
+             short_help='Checks that provided expression is present in input'
+                        ' file.',
```

Comment:
![image](https://user-images.githubusercontent.com/274668/140758134-9f6236a6-92a2-40a9-8f28-9d90e309d1d4.png)

How about we change this to:

```suggestion
             short_help='Assert expression in Result.',
```